scalariform/src/test/scala/scalariform/formatter/NewlineAtEndOfFileFormatterTest.scala

```diff
@@ -0,0 +1,41 @@
+package scalariform.formatter
+
+import scalariform.parser.{CompilationUnit, ScalaParser}
+import scalariform.formatter.preferences.{FormattingPreferences, NewlineAtEndOfFile}
+
+/** Tests that top-level parses add a newline when NewlineAtEndOfFile is set. */
+class NewlineAtEndOfFileFormatterTest extends AbstractFormatterTest {
+  type Result = CompilationUnit
+
+  // Must parse as a full script to verify the newline formatting.
+  def parse(parser: ScalaParser) = parser.scriptBody()
+
+  def format(formatter: ScalaFormatter, result: Result) = formatter.format(result)(FormatterState())
+
+  override val debug = false
+
+  {
+  implicit val formattingPreferences = FormattingPreferences.setPreference(NewlineAtEndOfFile, true)
```

Comment:
Note that I had to rename them due to how the test cases are generated - they need different text, else they end up with the same name & cause errors while running.